hoomd/HOOMDMath.h

```diff
@@ -349,6 +349,50 @@ inline HOSTDEVICE double pow(double x, double y)
     return ::exp(y * log(x));
     }
 
+//! Compute the pow of x,y in double precision when y is an integer using squaring
+inline HOSTDEVICE double pow(double x, unsigned int y){
+        double result = 1.0;
+        for(;;){
+                if(y & 1)
+                    result *= x;
+                y >>= 1;
+                if(!y)
```

Comment:
Even with all these branches, this method is faster than `::expf(y * logf(x))`? Did you benchmark on both the CPU and GPU? I would like to see performance metrics showing that the method is worth it.

Reply:
I briefly tested the best-case scenario on an A100, which is of monodisperse values for the exponent (µ=v=2). The integer version produces 4100 TPS for integers (vs 4300 for 12-6 LJ), and 2600 with FP exponents. Other values of exponents I tried do not give measurable differences in performance. 

I will have to inspect the PTX in more details, but I'm assuming nvcc generated branchless code, eg `result = y&1 ? x : 1.0` and a select op. Cost would be proportional to the log2 of the largest exponent in the warp. Exponents are generally small, say 1 ~ 4. Largest value in the force-field of interest (MPIPI) is 11.

Reply:
Thanks!

Reply:
I looked into the PTX, nvcc indeed yielded branchless code with FSEL for the the multiplication. The performance difference is even higher when compute bound; I tested on my old 2060, and got 225 TPS (int) versus 100 TPS (float).